docs/review.md

```diff
@@ -0,0 +1,579 @@
+  end
+```
+
+See https://github.com/makersacademy/course/blob/master/pills/spec_helper_methods.md
+
+
+## Ensure asset routes are set correctly
+
+Do we like http://recipes.sinatrarb.com/p/asset_management/sinatra_assetpack ?
+
+Or should we just be telling them to set the public folder correctly
+
+```ruby
+set :public_folder, Proc.new { File.join(root, 'static') }
+```
+
+and for Heroku:
+
+* http://abernardes.github.io/2014/10/24/static-assets-with-rack.html
+* http://stackoverflow.com/a/5509152/316729
+
+add the following to config.ru
+
+```ruby
+use Rack::Static, :urls => ['/stylesheets', '/javascripts'], :root => 'public'
+```
+
+# Step 4: Application code and \*.rb files
+
+## Controllers
+
+### Prefer Sinatra::Application vs Sinatra::Base
+
+TODO work out if we care
+
+* http://www.sinatrarb.com/intro.html#Modular%20vs.%20Classic%20Style
+
+### Be clear about how to use flash.now
+
+Have to register Sinatra Flash before using. Have to use
+
+```
+flash.next[:notice]
+```
+
+and not
+
+```
+flash.now[:notice]
+```
+
+in order to see the message after the redirect.
+
+### Refactor long controller methods
+
+This overly long controller method with business logic spread spread out throughout the method is not ideal.
+
+```ruby
+post '/password_reset' do
+  user = User.first(email: params[:email])
+  if user
+    user.reset_password
+    user.save
+    flash[:info] = 'Check your emails'
+    redirect('/')
+  else
+    flash[:info] = 'No such account on record'
+    redirect('/password_reset')
+  end
+end
+```
+
+It could be refactored like so:
+
+```ruby
+post '/password_reset' do
+  if User.reset_password(email: params[:email])
+    flash[:info] = 'Check your emails'
+    redirect('/')
+  else
+    flash[:info] = 'No such account on record'
+    redirect('/password_reset')
+  end
+end
+```
+
+Note the much shorter method and the business logic all pulled into the User model.
+
+### Split routes into a controller file/many controller files +
+
+Rather than
+
+```ruby
+class Chitter < Sinatra::Application
+  get '/' do
+    redirect '/peeps'
+  end
+
+  get '/peeps' do
+    @peeps = Peep.all
+    erb :'peeps/index'
+  end
+
+  get '/peeps/new' do
+    erb :'peeps/new'
+  end
+end
+```
+
+prefer
+
+```ruby
+# app.rb
+require 'controllers/peep_controller'
+
+class Chitter < Sinatra::Application
+  use Routes::PeepController
+end
+
+# controllers/peep_controller.rb
+module Routes
+  class PeepController < Sinatra::Application
+    get '/' do
+      redirect '/peeps'
+    end
+
+    get '/peeps' do
+      @peeps = Peep.all
+      erb :'peeps/index'
+    end
+
+    get '/peeps/new' do
+      erb :'peeps/new'
+    end
+  end
+end
+```
+
+TODO - do we need the `use` keyword here?
+TODO - watch out for unnecessary inheritance - does PeepController need to extend anything?
+
+## Models
+
+### Not encapsulating business logic in the model
+
+see 'refactor long controller methods' above where business logic is pulled from controller into User model
```

Comment:
we do do the breaking up like this, but in the curriculum we just monkey patched the Sinatra::Base class rather than using this `use` method.  I saw some students using this in their challenges last time - not clear if it is "better practice" or not ...